clang/test/OpenMP/task_ast_print.cpp

```diff
@@ -208,4 +209,39 @@ int main(int argc, char **argv) {
 extern template int S<int>::TS;
 extern template long S<long>::TS;
 
+// DUMP-LABEL:  FunctionDecl {{.*}} implicit_firstprivate
+void
+implicit_firstprivate() {
+
+#pragma omp parallel num_threads(1)
+  {
+    int i = 0;
+    // DUMP: OMPTaskDirective 
+    // DUMP-NEXT: OMPFirstprivateClause
+    // DUMP-NEXT: DeclRefExpr {{.+}} 'i' {{.+}} refers_to_enclosing_variable_or_capture
```

Comment:
This could still match if `non_odr_use_unevaluated` shows up, can't it? Can we add a check that `non_odr_use_unevaluated` should _not_ be part of the `DeclRefRexpr` under this firstprivate clause?